llvm/lib/Target/AArch64/AArch64Subtarget.cpp

```diff
@@ -584,6 +584,10 @@ AArch64Subtarget::getAuthenticatedLRCheckMethod() const {
   return AArch64PAuth::AuthCheckMethod::None;
 }
 
+bool AArch64Subtarget::enableSubRegLiveness() const {
```

Comment:
It is probably worth adding an option like in the Arm/RISCV backends too, so it can be disabled in case of problems.

Reply:
Or we could make the -enable-subreg-liveness option actually work and not require every target to do this

Reply:
I don't mind looking into this, if someone else does not already. It would be good to get this patch in though, in case we do find any issues before the clang-19 release.

Reply:
I've removed the `-aarch64-enable-subreg-liveness` flag from this patch, as the existing flag can be used to disable liveness tracking when enableSubRegLiveness just returns true.

Reply:
Thanks. I put up #95437 for trying to fix the option.